src/main/java/net/snowflake/client/jdbc/SnowflakeResultSetV1.java

```diff
@@ -50,6 +50,17 @@ public QueryStatus getStatus() throws SQLException {
     throw new SnowflakeLoggedFeatureNotSupportedException(session);
   }
 
+  /**
+   * This function is not supported for synchronous queries
+   *
+   * @return no return value; exception is always thrown
+   * @throws SQLFeatureNotSupportedException
+   */
+  @Override
+  public String getQueryErrorMessage() throws SQLException {
+    throw new SnowflakeLoggedFeatureNotSupportedException(session);
```

Comment:
Can we give explicit error message saying that feature not supported in synchronous mode. Only for async mode.

Reply:
sounds good! Added